actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -671,12 +675,18 @@ def use_relative_controller!
 
         # Remove leading slashes from controllers
         def normalize_controller!
-          @options[:controller] = controller.sub(%r{^/}, ''.freeze) if controller
+          if controller
+            if m = controller.match(/\A\/(?<controller_without_leading_slash>.*)/)
+              @options[:controller] = m[:controller_without_leading_slash]
+            else
+              @options[:controller] = controller
+            end
+          end
```

Comment:
While this may save on String allocations, I suspect the regex matching will actually make it slower. While the readability may be questionable, using `start_with?` and `[]` is much faster:

```rb
require 'benchmark/ips'

def sub(controller)
  controller.sub(%r{^/}, ''.freeze) if controller
end

def match(controller)
  if controller
    if m = controller.match(/\A\/(?<controller_without_leading_slash>.*)/)
      m[:controller_without_leading_slash]
    else
      controller
    end
  end
end

def start_with(controller)
  if controller
    if controller.start_with?('/'.freeze)
      controller[1..-1]
    else
      controller
    end
  end
end

Benchmark.ips do |x|
  x.report("sub") { sub("no_leading_slash") }
  x.report("match") { match("no_leading_slash") }
  x.report("start_with") { start_with("no_leading_slash") }

  x.compare!
end

Benchmark.ips do |x|
  x.report("sub") { sub("/a_leading_slash") }
  x.report("match") { match("/a_leading_slash") }
  x.report("start_with") { start_with("/a_leading_slash") }

  x.compare!
end
```
```
Calculating -------------------------------------
                 sub    59.229k i/100ms
               match    66.314k i/100ms
          start_with    91.586k i/100ms
-------------------------------------------------
                 sub      1.140M (± 9.0%) i/s -      5.686M
               match      1.423M (± 7.1%) i/s -      7.096M
          start_with      3.590M (±10.1%) i/s -     17.768M

Comparison:
          start_with:  3589743.1 i/s
               match:  1423380.7 i/s - 2.52x slower
                 sub:  1140204.8 i/s - 3.15x slower

Calculating -------------------------------------
                 sub    41.670k i/100ms
               match    37.536k i/100ms
          start_with    90.949k i/100ms
-------------------------------------------------
                 sub    653.775k (± 7.6%) i/s -      3.250M
               match    523.276k (±10.9%) i/s -      2.590M
          start_with      2.329M (±12.3%) i/s -     11.460M

Comparison:
          start_with:  2328795.8 i/s
                 sub:   653774.6 i/s - 3.56x slower
               match:   523276.2 i/s - 4.45x slower
```